tests/integration/test_monitor.py

```diff
@@ -0,0 +1,139 @@
+import os
+
+import numpy as np
+import pytest
+import tensorflow as tf
+
+import gpflow
+from gpflow.monitor import (
+    ModelToTensorBoard,
+    ScalarToTensorBoard,
+    ImageToTensorBoard,
+    MonitorCollection,
+)
+
+
+class Data:
+    num_data = 20
+    num_steps = 10
+
+
+@pytest.fixture
+def model():
+    data = (
+        np.random.randn(Data.num_data, 2),  # [N, 2]
+        np.random.randn(Data.num_data, 2),  # [N, 1]
+    )
+    kernel = gpflow.kernels.SquaredExponential(lengthscales=[1.0, 2.0])
+    model = gpflow.models.GPR(data, kernel, noise_variance=0.01)
+
+    return model
+
+
+def _get_size_directory(dir):
+    """ Calculating the size of a directory (in Bytes) """
+    return sum(d.stat().st_size for d in os.scandir(dir) if d.is_file())
+
+
+def test_logdir_created(model, tmp_path):
+    tmp_path = str(tmp_path)
+
+    def elbo_callback():
+        return model.log_likelihood()
+
+    monitor = MonitorCollection(
+        [ModelToTensorBoard(tmp_path, model), ScalarToTensorBoard(tmp_path, elbo_callback, "elbo"),]
+    )
+
+    # check existence
+    assert os.path.exists(tmp_path) and os.path.isdir(tmp_path)
+    size_before = _get_size_directory(tmp_path)
+    assert size_before > 0
+
+    @tf.function
+    def closure():
+        return -1.0 * model.log_likelihood()
+
+    opt = tf.optimizers.Adam()
+    for step in range(Data.num_steps):
+        opt.minimize(closure, model.trainable_variables)
+        monitor(step)
+
+    size_after = _get_size_directory(tmp_path)
+    assert size_after > size_before
```

Comment:
Does tensorflow write any error files when things go wrong? Would it be possible to explicitly check for the existence of the correct tensorboard file ? (I know there's some "randomness" in the path due to the timestamp, but could probably handle that with glob.glob()?)